samples/highcharts/blog/line-chart/demo.js

```diff
@@ -0,0 +1,168 @@
+                fontWeight: 'bold'
+            }
+        },
+        title: {
+            text: null,
+            style: {
+                color: '#FFFFFF',
+                fontWeight: 'bold'
+            }
+        }
+    },
+
+    xAxis: {
+        opposite: true,
+        offset: 10,
+        labels: {
+            style: {
+                color: '#FFFFFF',
+                fontWeight: 'bold'
+            }
+        },
+        accessibility: {
+            rangeDescription: 'Range: 2010 to 2020'
+        }
+    },
+
+    legend: {
+        align: 'right',
+        verticalAlign: 'middle',
+        layout: 'proximate',
+        itemStyle: {
+            color: '#FFFFFF',
+            fontWeight: 'bold'
+        }
+    },
+
+    plotOptions: {
+        series: {
+            label: {
+                connectorAllowed: false
+            },
+            marker: {
+                enabled: false,
+                symbol: 'circle'
+            },
+            pointStart: 1980,
+            pointInterval: 5,
+            lineWidth: 4
+        }
+    },
+    colors: [
+        '#FF0200',
+        '#7CB4EC',
+        '#008001',
+        '#faf0be',
+        '#FFB6C1',
+        '#2C908F',
+        '#91EE7D',
+        '#F7A35B',
+        '#8185E8',
+        '#F15C80',
+        '#ffffff',
+        '#A19642'
+    ],
+
+    series: [
+        {
+            name: 'China',
+            data: [10, 9, 7, 3, 2, 2, 2, 1, 1]
+        },
+        {
+            name: 'USA',
+            data: [1, 2, 1, 1, 1, 1, 1, 2, 2]
+        },
+        {
+            name: 'India',
+            data: [9, 10, 10, 5, 5, 4, 3, 3, 3]
+        },
+        {
+            name: 'Germany',
+            data: [4, 4, 4, 4, 4, 5, 5, 5, 5]
+        },
+        {
+            name: 'Japan',
+            data: [3, 3, 3, 2, 3, 3, 4, 4, 4]
+        },
+        {
+            name: 'Russia',
+            data: [null, null, null, 6, 7, 6, 6, 6, 6]
+        },
+        {
+            name: 'Indonesia',
+            data: [null, null, null, null, null, null, null, 8, 7]
+        },
+        {
+            name: 'Brazil',
+            data: [7, 7, 9, 9, 9, 7, 7, 7, 8]
+        },
+        {
+            name: 'France',
+            data: [6, 6, 6, 8, 6, 8, 8, 10, 9]
+        },
+        {
+            name: 'UK',
+            data: [8, 8, 8, 10, 10, 9, 9, 9, 10]
+        },
+        {
+            name: 'Soviet Union',
+            data: [2, 1, 2, null],
+            showInLegend: false
+        },
+        {
+            name: 'Italy',
+            data: [5, 5, 5, null],
+            showInLegend: false
+        }
+    ],
+
+    responsive: {
+        rules: [
+            {
+                condition: {
+                    maxWidth: 500
+                },
+                chartOptions: {
+                    legend: {
+                        layout: 'horizontal',
+                        align: 'center',
+                        verticalAlign: 'bottom'
+                    }
```

Comment:
This responsive rule actually doesn't work. First of all, it needs to set `chart.marginBottom` to `undefined` in order to have space for the legend. But even then, it seems HC is not able to switch from `layout: 'proximate'` to `layout: 'horizontal'`. That's a bug, and I'll investigate.

My suggestion is that we remove the responsive rule. The legend looks fine on mobile with the initial settings too in my opinion.

Reply:
Related issue https://github.com/highcharts/highcharts/issues/18362